x/exchange/types/escrow.go

```diff
@@ -38,9 +38,9 @@ func (e *Escrow) Unlock(addr sdk.AccAddress, amt ...sdk.DecCoin) {
 
 func (e *Escrow) Pays(addr sdk.AccAddress) sdk.Coins {
 	var pays sdk.Coins
-	for _, decCoin := range e.deltas[addr.String()] {
-		if decCoin.IsNegative() {
-			coin := sdk.NewCoin(decCoin.Denom, decCoin.Amount.Neg().Ceil().TruncateInt())
+	for _, coin := range e.deltas[addr.String()] {
+		if coin.IsNegative() {
+			coin.Amount = coin.Amount.Neg()
 			pays = pays.Add(coin)
 		}
```

Comment:
What about the case of IsPositive()? If it is not allowed, the error handling (or sanity check) is needed.

Reply:
Negative coin amount and positive coin amount are handled separately in `Pays` and `Receives`.